#[derive(Properties)] fails for some trait bounds

<!-- Thanks for filing a yew issue! We appreciate it :-) -->

#### Description

**I'm submitting a bug report.** 

With 0.8.0, an ability to auto-derive the `Properties` trait was introduced, but this fails to compile when certain trait bounds are specified. This is still true on current master (24d39f97e3e2818896b1338cb11b4db6eff76491).

#### Expected Results

This compiles without error:

```
use std::str::FromStr;

#[derive(Properties, Clone)]
pub struct Compiles<T: FromStr + Clone>
    where
        <T as FromStr>::Err: Clone,
{
    #[props(required)]
    value: Result<T, <T as FromStr>::Err>,
}
```

Note the trait bounds on `T` and on `<T as FromStr>::Err`. Namely, they are:

```
T: FromStr + Clone
<T as FromStr>::Err: Clone
```

This is expected. I would also expect that the auto-derive of Properties could handle the case where the trait bound in put in a where clause.

#### Actual Results

However, if I use this

```
use std::str::FromStr;

#[derive(Properties, Clone)]
pub struct FailsToCompile<T>
    where
        T: FromStr + Clone,
        <T as FromStr>::Err: Clone,
{
    #[props(required)]
    value: Result<T, <T as FromStr>::Err>,
}
```

I get the following compilation error:

```
error[E0277]: the trait bound `T: std::str::FromStr` is not satisfied
  --> examples/props_traits/src/lib.rs:31:10
   |
31 | #[derive(Properties, Clone)]
   |          ^^^^^^^^^^ the trait `std::str::FromStr` is not implemented for `T`
   |
   = help: consider adding a `where T: std::str::FromStr` bound

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: Could not compile `props_traits`.
```

The compiler suggests to add the trait bound `T: FromStr`, but this is already present. So it seems the derive macro fails to include the trait bound specified.

#### Context (Environment)

<!-- precise nightly or stable -->
- Rust: 1.37.0

<!-- if you're not on a tagged version precise the commit hash -->
- yew: 24d39f97e3e2818896b1338cb11b4db6eff76491

--- a/crates/macro/src/derive_props/builder.rs
+++ b/crates/macro/src/derive_props/builder.rs
@@ -64,7 +64,9 @@ impl ToTokens for PropsBuilder<'_> {
             #(impl #step_trait for #step_names {})*
 
             #[doc(hidden)]
-            #vis struct #builder_name#step_generics {
+            #vis struct #builder_name#step_generics
+            #where_clause
+            {
                 wrapped: ::std::boxed::Box<#wrapper_name#ty_generics>,
                 _marker: ::std::marker::PhantomData<#step_generic_param>,
             }

--- a/crates/macro/src/derive_props/builder.rs
+++ b/crates/macro/src/derive_props/builder.rs
@@ -65,7 +65,7 @@ impl ToTokens for PropsBuilder<'_> {
 
             #[doc(hidden)]
             #vis struct #builder_name#step_generics
-            #where_clause
+                #where_clause
             {
                 wrapped: ::std::boxed::Box<#wrapper_name#ty_generics>,
                 _marker: ::std::marker::PhantomData<#step_generic_param>,

--- a/crates/macro/src/derive_props/wrapper.rs
+++ b/crates/macro/src/derive_props/wrapper.rs
@@ -24,7 +24,9 @@ impl ToTokens for PropsWrapper<'_> {
         let wrapper_default_setters = self.default_setters();
 
         let wrapper = quote! {
-            struct #wrapper_name#generics {
+            struct #wrapper_name#generics
+                #where_clause
+            {
                 #(#wrapper_field_defs)*
             }
 

--- a/tests/derive_props/pass.rs
+++ b/tests/derive_props/pass.rs
@@ -83,4 +83,22 @@ mod t5 {
     }
 }
 
+mod t6 {
+    use super::*;
+    use std::str::FromStr;
+
+    #[derive(Properties, Clone)]
+    pub struct Props<T: FromStr + Clone>
+    where
+    <T as FromStr>::Err: Clone,
+    {
+        #[props(required)]
+        value: Result<T, <T as FromStr>::Err>,
+    }
+
+    fn required_prop_generics_with_where_clause_should_work() {
+        Props::<String>::builder().value(Ok(String::from(""))).build();
+    }
+}
+
 fn main() {}

